packages/viewer/src/components/viewer-box-query-tool/viewer-box-query-tool.spec.tsx

```diff
@@ -216,6 +217,91 @@ describe('vertex-viewer-box-query-tool', () => {
       })
     );
   });
+
+  it('disables the interaction handler while an operation is being executed', async () => {
+    const { stream, ws } = makeViewerStream();
+
+    const page = await newSpecPage({
+      components: [Viewer, ViewerBoxQueryTool, ViewerLayer],
+      template: () => (
+        <vertex-viewer stream={stream}>
+          <vertex-viewer-box-query-tool></vertex-viewer-box-query-tool>
+        </vertex-viewer>
+      ),
+    });
+
+    const viewer = page.root as HTMLVertexViewerElement;
+
+    await loadViewerStreamKey(key1, { viewer, stream, ws });
+
+    const streamSpy = jest
+      .spyOn(stream, 'createSceneAlteration')
+      .mockImplementation(async () => {
+        await new Promise((resolve) => setTimeout(resolve, 10));
+        return {};
+      });
+
+    await drawInclusiveBox(page, viewer);
+    window.dispatchEvent(new MouseEvent('pointerup'));
+
+    await drawExclusiveBox(page, viewer);
+    window.dispatchEvent(new MouseEvent('pointerup'));
+
+    await page.waitForChanges();
+    await Async.delay(10);
```

Comment:
Does the test fail without a manual delay?

Reply:
Yeah, unfortunately. At least the first delay is necessary to "complete" the first alteration and allow for the third attempt to complete successfully. Looks like I'm able to remove the second await though, so I'll do that